src/daft-local-execution/src/progress_bar.rs

```diff
@@ -0,0 +1,84 @@
+use std::{borrow::Cow, sync::atomic::AtomicU64};
+
+use indicatif::{HumanCount, ProgressBar, ProgressStyle};
+
+pub enum ProgressBarColor {
+    Blue,
+    Magenta,
+    Cyan,
+}
+
+impl ProgressBarColor {
+    fn to_str(&self) -> &'static str {
+        match self {
+            Self::Blue => "blue",
+            Self::Magenta => "magenta",
+            Self::Cyan => "cyan",
+        }
+    }
+}
+
+pub struct OperatorProgressBar {
+    inner_progress_bar: ProgressBar,
+    emitted: AtomicU64,
```

Comment:
would I would recommend instead is we pass in the `RuntimeStatsRef` here and then just call a `redraw` or `render` method instead. That way we only ever have to update 1 atomic.

Reply:
Makes sense, implemented.